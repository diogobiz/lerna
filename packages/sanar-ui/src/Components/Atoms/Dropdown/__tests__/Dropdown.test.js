import React from "react";
import TestRenderer from "react-test-renderer";

import ESDropdown, { ESDropdownButton }  from "../";
import ESMenu, { ESItem, ESDivider } from "../../Menu";
import ESButton from "../../Button";

const menu = (
	<ESMenu>
		<ESItem key="1">Item 1</ESItem>
		<ESItem key="2">Item 2</ESItem>
		<ESDivider />
		<ESItem key="3">Item 3</ESItem>
	</ESMenu>
);

it("renders correctly", () => {	
	const component = (
		<ESDropdown overlay={menu}>
			<ESButton>Dropdown</ESButton>
		</ESDropdown>
	)

	const tree = TestRenderer.create(component).toJSON();
	expect(tree).toMatchSnapshot();
});

it("renders correctly", () => {	
	const component = (
		<ESDropdownButton overlay={menu}>
			Dropdown
		</ESDropdownButton>
	)

	const tree = TestRenderer.create(component).toJSON();
	expect(tree).toMatchSnapshot();
});
