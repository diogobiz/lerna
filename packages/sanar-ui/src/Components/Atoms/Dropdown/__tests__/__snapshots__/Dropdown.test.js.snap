// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="es-button ant-dropdown-trigger es-dropdown es-button__medium ant-btn"
  data-testid="es-button"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  <span>
    Dropdown
  </span>
</button>
`;

exports[`renders correctly 2`] = `
<div
  className="ant-btn-group ant-dropdown-button es-dropdown-button"
>
  <button
    className="ant-btn ant-btn-default"
    onClick={[Function]}
    type="button"
  >
    <span>
      Dropdown
    </span>
  </button>
  <button
    className="ant-btn ant-dropdown-trigger ant-btn-default"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  >
    <i
      aria-label="icon: ellipsis"
      className="anticon anticon-ellipsis"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="ellipsis"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M176 511a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm280 0a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm280 0a56 56 0 1 0 112 0 56 56 0 1 0-112 0z"
        />
      </svg>
    </i>
  </button>
</div>
`;
